### ISSUE TITLE
POS App set to root domain issue

### ISSUE THREAD

**Issue by britttttk** [btcpayserver/btcpayserver#1694]
**Describe the problem/bug**
Setting policy > display POS app on website root gives 500 (for the root domain).

**Your environment**
* Version of BTCPay Server:  v1.0.5.0
* Deployment method: LunaNode

**Logs (if applicable)**
```
2020-06-27 01:04:04.030 +00:00 [INF] BTCPayServer.Services.PoliciesSettings
2020-06-27 01:04:12.722 +00:00 [ERR] The view '/Views/AppsPublic/ViewPointOfSale.cshtml' was not found. Searched locations: ["/Views/AppsPublic/ViewPointOfSale.cshtml"]
2020-06-27 01:04:12.800 +00:00 [ERR] Connection id "0HM0Q6NB80FFC", Request id "0HM0Q6NB80FFC:00000001": An unhandled exception was thrown by the application.
System.InvalidOperationException: The view '/Views/AppsPublic/ViewPointOfSale.cshtml' was not found. The following locations were searched:
/Views/AppsPublic/ViewPointOfSale.cshtml
   at Microsoft.AspNetCore.Mvc.ViewEngines.ViewEngineResult.EnsureSuccessful(IEnumerable`1 originalLocations)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.ViewResultExecutor.ExecuteAsync(ActionContext context, ViewResult result)
   at Microsoft.AspNetCore.Mvc.ViewResult.ExecuteResultAsync(ActionContext context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResultFilterAsync>g__Awaited|29_0[TFilter,TFilterAsync](ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResultExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.ResultNext[TFilter,TFilterAsync](State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeResultFilters()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Session.SessionMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Session.SessionMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at BTCPayServer.Hosting.BTCPayMiddleware.Invoke(HttpContext httpContext) in /source/BTCPayServer/Hosting/BTCpayMiddleware.cs:line 87
   at Microsoft.AspNetCore.Diagnostics.StatusCodePagesMiddleware.Invoke(HttpContext context)
   at BTCPayServer.Hosting.HeadersOverrideMiddleware.Invoke(HttpContext httpContext) in /source/BTCPayServer/Hosting/HeadersOverrideMiddleware.cs:line 31
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpAp
```

**To Reproduce**
1. Go to: Server Settings > Policies
2. Click on: Display app on website root
3. Select from dropdown: Any POS app
4. Click Save
5. Open browser to root domain to see POS App
6. Page gives 500 error

**Expected behavior**
POS App appears on the root domain

**Actual behavior**
500 error shows for root domain

**Additional context**
Setting a Crowdfund App on root domain does not have the issue

**Comment by dennisreimann** [comment]
This is related to the refactoring I did. Will take a look.

**Comment by britttttk** [comment]
@dennisreimann I believe it was [fixed already](https://github.com/btcpayserver/btcpayserver/commit/16f2ddddf0751b87a15eff306bbbbc2b158782c3).

**Comment by britttttk** [comment]
yes confirmed, closing this now.

**Comment by dennisreimann** [comment]
Ah ok, thanks! 👍

**Comment by britttttk** [comment]
@dennisreimann @NicolasDorier on v1.0.5.1 now pos is fixed but crowdfund does not work for me

**Comment by NicolasDorier** [comment]
fixed in 1.0.5.2
